bagitobjecttransfer/recordtransfer/tests/unit/test_utils.py

```diff
@@ -109,123 +127,326 @@ class FileCountingUtilityTests(TestCase):
-            [],
-            self.accepted_formats,
-            logger=self.null_logger
+            [], self.accepted_formats, logger=self.null_logger
         )
-        self.assertEqual(statement, 'No file types could be identified')
+        self.assertEqual(statement, "No file types could be identified")
 
     def test_no_accepted_file_types_human_readable(self):
         statement = get_human_readable_file_count(
-            ['file1.pages', 'file2.docs', 'file3.pdf'],
+            ["file1.pages", "file2.docs", "file3.pdf"],
             self.accepted_formats,
-            logger=self.null_logger
+            logger=self.null_logger,
         )
-        self.assertEqual(statement, 'No file types could be identified')
+        self.assertEqual(statement, "No file types could be identified")
 
     def test_one_file_group_singular_human_readable(self):
         statement = get_human_readable_file_count(
-            ['file1.DOC'],
-            self.accepted_formats,
-            logger=self.null_logger
+            ["file1.DOC"], self.accepted_formats, logger=self.null_logger
         )
-        self.assertEqual(statement, '1 Microsoft Word Document file')
+        self.assertEqual(statement, "1 Microsoft Word Document file")
 
     def test_one_file_group_plural_human_readable(self):
         statement = get_human_readable_file_count(
-            ['file1.DOC', 'file2.doc', 'file3.docx', 'file4.DOCX'],
+            ["file1.DOC", "file2.doc", "file3.docx", "file4.DOCX"],
             self.accepted_formats,
-            logger=self.null_logger
+            logger=self.null_logger,
         )
-        self.assertEqual(statement, '4 Microsoft Word Document files')
+        self.assertEqual(statement, "4 Microsoft Word Document files")
 
     def test_two_groups_human_readable(self):
         statement = get_human_readable_file_count(
-            ['file1.DOC', 'file2.doc', 'song1.mp3', 'img1.jpg', 'img2.png', 'img3.png'],
+            ["file1.DOC", "file2.doc", "song1.mp3", "img1.jpg", "img2.png", "img3.png"],
             self.accepted_formats,
-            logger=self.null_logger
+            logger=self.null_logger,
         )
-        self.assertIn('2 Microsoft Word Document files', statement)
-        self.assertIn('1 Audio file', statement)
-        self.assertIn('3 Image files', statement)
+        self.assertIn("2 Microsoft Word Document files", statement)
+        self.assertIn("1 Audio file", statement)
+        self.assertIn("3 Image files", statement)
+
+
+class TestAcceptFile(TestCase):
+    """Tests for accept_file method."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Set up class-level test data."""
+        super().setUpClass()
+        settings.ACCEPTED_FILE_FORMATS = {"Document": ["docx", "pdf"], "Spreadsheet": ["xlsx"]}
+        settings.MAX_TOTAL_UPLOAD_SIZE = 3  # MiB
+        settings.MAX_SINGLE_UPLOAD_SIZE = 1  # MiB
+        settings.MAX_TOTAL_UPLOAD_COUNT = 4  # Number of files
+
```

Comment:
I recommend patching these settings in this class instead:

```suggestion

@patch(
    "django.conf.settings.ACCEPTED_FILE_FORMATS",
    {"Document": ["docx", "pdf"], "Spreadsheet": ["xlsx"]},
)
@patch("django.conf.settings.MAX_TOTAL_UPLOAD_SIZE", 3)  # MiB
@patch("django.conf.settings.MAX_SINGLE_UPLOAD_SIZE", 1)  # MiB
@patch("django.conf.settings.MAX_TOTAL_UPLOAD_COUNT", 4)  # Number of files
class TestAcceptFile(TestCase):
    """Tests for accept_file method."""

```